crates/pulsar-core/src/pdk/module.rs

```diff
@@ -18,11 +18,13 @@ impl<'a> TryFrom<&'a ModuleConfig> for NoConfig {
     }
 }
 
-/// Trait to implement to create a pulsar pluggable module
+/// Trait to implement to create a pulsar pluggable module. Note that this is the fully
+/// featured interface which is often too much. Please see [`BasicPulsarModule`] for a simpler interface.
 pub trait PulsarModule: Send {
     type Config: for<'a> TryFrom<&'a ModuleConfig, Error = ConfigError> + Send + Sync + 'static;
     type State: Send + 'static;
-    type Extra: Extra<Self>;
+    type ExtraState: Send + 'static;
```

Comment:
Since we are getting rid of "extra" naming, how do you feel about renaming `State` to `BasicState` and `ExtraState` to `State` (or `PulsarState`)?

Reply:
I ended up renaming `ExtraState` to `Extension`. Idk if `state` is necessarily the best name and ultimately I'd like to keep the main state named `State`. Let me know what you think.